include/ck/library/utility/device_tensor_generator.hpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Advanced Micro Devices, Inc., or its affiliates.
+// SPDX-License-Identifier: MIT
+#pragma once
+#include <hip/hip_runtime.h>
+
+#include "ck/ck.hpp"
+#include "ck/utility/common_header.hpp"
+#include "ck/library/utility/device_tensor_generator.hpp"
+#include "ck/utility/data_type.hpp"
+#include <cmath>
+
+// use xorshift for now since it is simple. Should be suitable enough, but feel free to switch in
+// the future
+struct ran_state_u32
+{
+    uint32_t s[4];
+};
+
+__device__ uint32_t ran_gen_round_u32(ran_state_u32& state)
+{
+    uint32_t tmp = state.s[3];
+    state.s[3]   = state.s[2];
+    state.s[2]   = state.s[1];
+    state.s[1]   = state.s[0];
+    tmp ^= tmp << 11;
+    tmp ^= tmp >> 8;
+    state.s[0] = tmp ^ state.s[0] ^ (state.s[0] >> 19);
+    return state.s[0];
+}
+
+__device__ ran_state_u32 ran_init(uint32_t seed = 0)
+{
+    ran_state_u32 state;
+    // use primes for initialization
+    state.s[0] = (blockDim.x * blockIdx.x + threadIdx.x) * 8912741 + 2313212 + seed;
+    state.s[1] =
+        (gridDim.x * blockDim.x - (blockDim.x * blockIdx.x + threadIdx.x)) * 5013829 + 6012697;
+    state.s[2] = (blockDim.x * blockIdx.x + threadIdx.x) * 3412309 + 2912479;
+    state.s[3] =
+        (gridDim.x * blockDim.x - (blockDim.x * blockIdx.x + threadIdx.x)) * 1001447 + 9912307;
+
+    // run 20 rounds
+    for(int i = 0; i < 20; i++)
+    {
+        ran_gen_round_u32(state);
+    }
+    return state;
+}
+
+template <typename T>
+__global__ void fill_tensor_uniform_rand_int_values(T* p,
+                                                    int min_value,
+                                                    int max_value,
+                                                    uint64_t buffer_element_size)
+{
+    // initial values
+    ran_state_u32 s = ran_init();
+    for(uint64_t i = blockIdx.x * blockDim.x + threadIdx.x;
+        i < buffer_element_size / ck::packed_size_v<T>;
+        i += blockDim.x * gridDim.x)
+    {
+        if constexpr(ck::is_same_v<T, ck::pk_i4_t>)
+        {
+            uint8_t hi      = ((ran_gen_round_u32(s)) % (max_value - min_value)) + min_value + 8;
+            uint8_t lo      = ((ran_gen_round_u32(s)) % (max_value - min_value)) + min_value + 8;
```

Comment:
This generates a bias. See https://romailler.ch/2020/07/28/crypto-modulo_bias_guide/.